core/templates/components/summary-tile/topic-summary-tile.component.ts

```diff
@@ -35,13 +38,52 @@ export class TopicSummaryTileComponent {
   constructor(
     private urlInterpolationService: UrlInterpolationService) {
   }
-  getTopicLink(): string {
-    return this.urlInterpolationService.interpolateUrl(
-      ClassroomDomainConstants.TOPIC_VIEWER_URL_TEMPLATE, {
-        topic_name: this.topicSummary.getName()});
+  openTopicPage(): string {
+    window.open(
+      this.urlInterpolationService.interpolateUrl(
+        ClassroomDomainConstants.TOPIC_VIEWER_URL_TEMPLATE, {
+          topic_name: this.topicSummary.getName()}), '_blank');
+  }
+  getColorValueInHexForm(colorValue: number): string {
+    colorValue = (colorValue < 0) ? 0 : colorValue;
+    var colorValueString = colorValue.toString(16);
+    return (
+      (colorValueString.length === 1) ?
+      '0' + colorValueString : colorValueString);
+  }
+  getDarkerThumbnailBgColor(): string {
+    var bgColor = this.topicSummary.getThumbnailBgColor();
+    // Remove the '#' from the first position.
+    bgColor = bgColor.slice(1);
+
+    // Get RGB values of new darker color.
+    var newRValue = this.getColorValueInHexForm(
+      parseInt(bgColor.substring(0, 2), 16) - 100);
+    var newGValue = this.getColorValueInHexForm(
+      parseInt(bgColor.substring(2, 4), 16) - 100);
+    var newBValue = this.getColorValueInHexForm(
+      parseInt(bgColor.substring(4, 6), 16) - 100);
+
+    return '#' + newRValue + newGValue + newBValue;
   }
-  getStaticImageUrl(imagePath: string): string {
-    return this.urlInterpolationService.getStaticImageUrl(imagePath);
+  getThumbnailUrl(): string {
+    // Since an Angular8 component cannot depend on an AngularJS service,
+    // the thumbnail URL is manually built here. This can be reverted to use
+    // assets backend api service once that is migrated to Angular 8.
+    // The constants used are copied over from AssetsBackendApiService.
```

Comment:
Is this fine as a temporary fix for now? Should I create an issue  for it?

Reply:
@aks681 who is your question addressed to?

Reply:
Ah sorry, /cc @bansalnitish @srijanreddy98. Can you take a look and see if what I've done for now is fine until `AssetsBackendApiService` is migrated to Angular 8?

Reply:
Resolving this since they have been converted back to directives in develop.